### ISSUE TITLE
The BoundServiceAccountTokenVolume feature creates tokens only readable by root

### ISSUE THREAD

**Issue by jsmilani** [kubernetes/kubernetes#78816]
**What happened**:
I am attempting to use the rotating service account certificate feature in kubernetes 1.14.3 and my infrastructure rotates the certificates every day but I found too many problems to fully implement.

This bug specifically addresses the permissions on the token file created inside the projected volume used by the BoundServiceAccountTokenVolume feature flag.

    $ ls -al /var/run/secrets/kubernetes.io/serviceaccount/token
    lrwxrwxrwx 1 root root 12 Jun  8 17:41 /var/run/secrets/kubernetes.io/serviceaccount/token -> ..data/token
    $ ls -al /var/run/secrets/kubernetes.io/serviceaccount/..data/token 
    -rw------- 1 root root 862 Jun  8 17:41 /var/run/secrets/kubernetes.io/serviceaccount/..data/token

As you can see the original token location has the correct permissions but that is a symlink to ..data/token which is only readable by root. This breaks many services that don't run as root (eg. tiller which runs as nobody).

**What you expected to happen**:

I would expect the ..data/token file to have at read  '-rw-r--r--' permissions.

**How to reproduce it (as minimally and precisely as possible)**:

On a clean kubernetes 1.14.3 installation...

create /tmp/kubeadm.conf:

    kind: ClusterConfiguration
    apiVersion: kubeadm.k8s.io/v1beta1
    apiServer:
      extraArgs:
        feature-gates: BoundServiceAccountTokenVolume=true
        service-account-issuer: api
        service-account-signing-key-file: /etc/kubernetes/pki/sa.key
        service-account-api-audiences: api
    controllerManager:
      extraArgs:
        feature-gates: "BoundServiceAccountTokenVolume=true"
    scheduler: {}
    ---
    kind: KubeletConfiguration
    apiVersion: kubelet.config.k8s.io/v1beta1
    featureGates:
      BoundServiceAccountTokenVolume: true

Init kubernetes master:

    kubeadm init --config=/tmp/kubeadm.conf

Locate the kube-proxy container:

    docker ps | grep k8s_kube-proxy

Inspect the container:

    docker inspect [CONTAINER]

Locate the Mounts section and find the "source" of the /var/run/secrets/kubernetes.io/serviceaccount directory. Then do an 'ls -al' on the "${SOURCE_DIR}/..data/" directory. Instead of '-rw-r--r--' the permissions will be '-rw-------'. If you repeat this process but without the config file I provided then the permissions will be '-rw-r--r--' which works correctly for non-root user containers.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version 1.14.3
- Cloud provider or hardware configuration: na
- OS (e.g: `cat /etc/os-release`): centos 7
- Kernel (e.g. `uname -a`): Linux  3.10.0-957.1.3.el7.x86_64 #1 SMP Thu Nov 29 14:49:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: na
- Network plugin and version (if this is a network-related bug): na
- Others:

**Comment by jsmilani** [comment]
@kubernetes/sig-auth

**Comment by liggitt** [comment]
Duplicate of https://github.com/kubernetes/kubernetes/issues/74565?

rw------- permissions are intentional. I would expect the owner of the file to be the user of the container.

**Comment by jsmilani** [comment]
@liggitt you are right about being duplicate of #74565. I searched forever trying to find more details about using BoundServiceAccountTokenVolume and found very little and what I found was scattered all over. That ticket at points to a fix for this token permission issue when configuring the pod/container spec, but unfortunately I am using it with tools such as helm/tiller which defines its own deployment. That alone is not a dealbreaker since I could work around it by managing the deployment myself, but it also heavily depends on third parties containers to adopt the token reload feature first and so far I have observed a great deal of instability with coredns and romana network. I already decided to postpone our implementation of this feature and am creating bugs and enhancement tickets for when I revisit it.

Closing this as duplicate.
